src/Illuminate/Database/Eloquent/Relations/HasOne.php

```diff
@@ -7,13 +7,41 @@
 class HasOne extends HasOneOrMany
 {
     /**
+     * Determine whether getResults should return a default new model instance or not.
+     *
+     * @var bool
+     */
+    protected $withDefault = false;
+
+    /**
+     * Return a new model instance in case the relationship does not exist.
+     *
+     * @return $this
+     */
+    public function withDefault()
+    {
+        $this->withDefault = true;
+
+        return $this;
+    }
+
+    /**
      * Get the results of the relationship.
      *
      * @return mixed
      */
     public function getResults()
     {
-        return $this->query->first();
+        if ($result = $this->query->first()) {
+            return $result;
+        }
+
+        if ($this->withDefault) {
+            return $this->related->newInstance()
+                ->setAttribute(
+                    $this->getPlainForeignKey(), $this->getParentKey()
+                );
+        }
     }
```

Comment:
if you don't return anything, the method will return `null` by default. So there is an implicit `return null;` at the end of the method.